underdocs-error/src/main/kotlin/underdocs/error/pretty/DefaultErrorPrettyPrinter.kt

```diff
@@ -1,7 +1,34 @@
 package underdocs.error.pretty
 
+import underdocs.error.BaseUnderdocsError
+
 class DefaultErrorPrettyPrinter : ErrorPrettyPrinter {
     override fun print(exception: Exception): String {
-        TODO("Not yet implemented")
+        var prettyPrintedError = """"""
+
+        if (exception is BaseUnderdocsError) {
+            prettyPrintedError = """
+                |${exception.title.toUpperCase()} - Error Code ${exception.code}
+                |Details
+                |${exception.getDescription()}
+                |
+                |Possible solution
+                |${exception.getSolution()}
+            """.trimMargin()
+
+            if (exception.cause != null) prettyPrintedError += "\nError cause:\n${exception.cause?.message}"
+        } else {
+            val errorStackTrace = exception.stackTrace.joinToString("\n\t", prefix = "\t")
```

Comment:
This is 100% better than a `for` loop. For me, it's perfect. Of course, it will use `StackTraceElement.toString()` so it's gonna display well, so no problems here.